events/ratelimiting/ratelimiter_test.go

```diff
@@ -0,0 +1,21 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ratelimiting
+
+import "testing"
+
+func TestRateLimiters(*testing.T) {
+	var _ RateLimiter = new(coalescing)
```

Comment:
Change to a compile-time assertion in `ratelimiter.go`

```go
var _ RateLimiter = (*coalescing)(nil)
// In unit.go
var _ RateLimiterWithTicker = (*coalescing)(nil)
```

(These don't cause allocations)

Reply:
Question: What is the benefit of having this in the compiled file? The benefit of having it in the test file is that it doesn't clutter the implementation code.

Reply:
I think it's just the pattern I've generally seen. It makes errors immediately visible by the compiler (it is also something that should have no or negligible impact on the compiled code) 🤷‍♂️

Also, would fix the errors the linter is reporting :)

Reply:
The linter is erroring because we are using a GitHub native action to run the linter rather than the Makefile, so the `unit` build tag is not being applied. I have opened https://github.com/dapr/kit/pull/67